cmd/helm/repo_add_test.go

```diff
@@ -101,3 +106,112 @@ func TestRepoAdd(t *testing.T) {
 		t.Errorf("Duplicate repository name was added")
 	}
 }
+func TestRepoAddConcurrentGoRoutines(t *testing.T) {
+	ts, thome, err := repotest.NewTempServer("testdata/testserver/*.*")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cleanup := resetEnv()
+	defer func() {
+		ts.Stop()
+		os.RemoveAll(thome.String())
+		cleanup()
+	}()
+
+	settings.Home = thome
+	if err := ensureTestHome(settings.Home, t); err != nil {
+		t.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(3)
+	for i := 0; i < 3; i++ {
+		go func(name string) {
+			// TODO: launch repository additions in sub-processes as file locks are bound to processes, not file descriptors
```

Comment:
Are the file locks bound to processes, threads, goroutines, or something else? goroutines are scheduled on top of threads and not guaranteed to be on different ones.

Reply:
This comment is obsolete, it comes from my first unix only implementation: https://github.com/helm/helm/pull/5678/commits/7ce22a3f585b574a2503de93a3f80ec75c1e5afb

github.com/gofrs/flock takes care of in process goroutine synchronization using mutexes in addition to using system level locks on files.

=> comment to remove.